### ISSUE TITLE
Account for deeply nested apps in basePath.

### ISSUE THREAD

**Issue by willist** [kogosoftwarellc/express-openapi#40]
When express-openapi is mounted on a sub app, the basePath ignores the mount path of the parent app.
```javascript
var app = express(),
    sub = express(),
    sub2 = express();

openapi.initialize({
  apiDoc: require('./api-doc.js'),
  app: sub2,
  routes: './api-routes'
});

sub.use('/subapp2', sub2);
app.use('/subapp', sub);
```

If ``./api-doc.js`` has a basePath of ``/v1``, I would expect the basePath in the generated api doc in this case to be ``/subapp/subapp2/v1``. Instead, the basePath of the api doc is ``/subapp2/v1``, ignoring the highest mount path.

**Comment by jsdevel** [comment]
@willist can you submit a PR for this?

**Comment by willist** [comment]
Fixed by #41

**Comment by jsdevel** [comment]
Thanks for closing!
